ci/infra/testrunner/tests/utils.py

```diff
@@ -31,7 +37,11 @@ def _handle_timeout(signum, frame):
         signal.signal(signal.SIGALRM, _handle_timeout)
         signal.alarm(timeout)
         try:
-            return func(*args, **kwargs)
+            start = timer()
```

Comment:
should we create a contextmanager to track time if we are gonna be using this down the line?

Reply:
I added a `contextmanager` that just yields and allow executing inside the block.

Thanks for the suggestion!